cli/src/tasks/init.ts

```diff
@@ -132,18 +124,68 @@ async function getWebDir(config: Config, webDir?: string) {
   return webDir;
 }
 
+async function runMergeConfig(
+  config: Config,
+  extConfig: ExternalConfig,
+  type: 'json' | 'ts',
+) {
+  const configDirectory = dirname(config.app.extConfigFilePath);
+  const newConfigPath = resolve(
+    configDirectory,
+    type === 'ts' ? CONFIG_FILE_NAME_TS : CONFIG_FILE_NAME_JSON,
+  );
+
+  await runTask(
+    `Creating ${c.strong(basename(newConfigPath))} in ${c.input(
+      config.app.rootDir,
+    )}`,
+    async () => {
+      await mergeConfig(config, extConfig, newConfigPath);
+    },
+  );
+
+  if (
+    newConfigPath !== config.app.extConfigFilePath &&
+    (await pathExists(config.app.extConfigFilePath))
+  ) {
+    const answers = await logPrompt(
+      `${c.strong(`Remove old ${config.app.extConfigName} file?`)}\n` +
+        `The path to your configuration file changed. Would you like to delete the old configuration file?`,
+      {
+        type: 'confirm',
+        name: 'confirm',
+        message: `Delete ${c.strong(config.app.extConfigName)}?`,
+        initial: true,
+      },
+    );
+
+    if (answers.confirm) {
+      await unlink(config.app.extConfigFilePath);
+    }
+  }
+
+  printNextSteps(basename(newConfigPath));
+}
+
 async function mergeConfig(
   config: Config,
   extConfig: ExternalConfig,
+  newConfigPath: string,
 ): Promise<void> {
   const oldConfig = { ...config.app.extConfig };
+  const newConfig = { ...oldConfig, ...extConfig };
 
-  await writeJSON(
-    config.app.extConfigFilePath,
-    {
-      ...oldConfig,
-      ...extConfig,
-    },
-    { spaces: 2 },
+  await writeConfig(newConfig, newConfigPath);
+}
+
+function printNextSteps(newConfigName: string) {
+  logSuccess(`${c.strong(newConfigName)} created!`);
+  output.write(
+    `\nAdd platforms using ${c.input('npx cap add')}:\n` +
+      `  ${c.input('npx cap add android')}\n` +
+      `  ${c.input('npx cap add ios')}\n\n` +
```

Comment:
Not related to this change since you just moved it, but, since platforms are not automatically installed anymore, this commands will end up in errors, so maybe we should remove it or reword?

Reply:
Sure, I think we should remove it and keep the link to the workflow docs below. We can add a section about adding platforms there.

Reply:
https://github.com/ionic-team/capacitor/issues/4030